### DISCUSSION TITLE
A few newbie iPXE questions...

### DISCUSSION THREAD

**Discussion by SuneThomsenDK** [ipxe/ipxe#322] [Q&A]
Hi,

I'm testing iPXE and I've got a few newbie questions ;)

The test system:
- Windows 2019 DC/DHCP
- Windows 2019 Webserver (IIS)
- Hyper-V Client 1 (BIOS Legacy)
- Hyper-V Client 2 (UEFI)

1. Does HTTPClient boot work in Hyper-V? DHCP Option 66 pointing towards http://webserver/boot.ipxe does not work for me.

2. So, I've installed the WDS role on the webserver and did all the chainloading and placed the undionly.kpxe / ipxe.efi in the "C:\RemoteInstall\Boot\x64" folder and got everything to work except Secure Boot on the UEFI Hyper-V client. And I have downloaded latest wimboot file "I think" version 2.7.2 which should have be signed, right?

So what´s my issue here? is it because that the ipxe.efi isen't sigend?

Thanks for this awesome project :D

**Comment by mcb30** [comment]
>     1. Does HTTPClient boot work in Hyper-V? DHCP Option 66 pointing towards http://webserver/boot.ipxe does not work for me.

No idea, sorry.  UEFI's built in HTTP boot support requires you to choose to HTTP boot within the UEFI setup UI (rather than just having the DHCP server hand out an `http://` URI), so take a look there - if you don't see a "UEFI HTTP boot" or similar option then it's not supported.

>     2. So, I've installed the WDS role on the webserver and did all the chainloading and placed the undionly.kpxe / ipxe.efi in the "C:\RemoteInstall\Boot\x64" folder and got everything to work except Secure Boot on the UEFI Hyper-V client. And I have downloaded latest wimboot file "I think" version 2.7.2 which should have be signed, right?
> 
> So what´s my issue here? is it because that the ipxe.efi isen't sigend?

Yes: the wimboot binary is signed, but iPXE isn't.

**Reply by SuneThomsenDK** [reply]
Thanks for the answers :) Do you know if they are working on getting the ipxe.efi signed?

**Comment by JBWinnipeg** [comment]
> 
> 
> Hi,
> 
> I'm testing iPXE and I've got a few newbie questions ;)
> 
> The test system:
> 
>     * Windows 2019 DC/DHCP
> 
>     * Windows 2019 Webserver (IIS)
> 
>     * Hyper-V Client 1 (BIOS Legacy)
> 
>     * Hyper-V Client 2 (UEFI)
> 
> 
>     1. Does HTTPClient boot work in Hyper-V? DHCP Option 66 pointing towards http://webserver/boot.ipxe does not work for me.
> 
>     2. So, I've installed the WDS role on the webserver and did all the chainloading and placed the undionly.kpxe / ipxe.efi in the "C:\RemoteInstall\Boot\x64" folder and got everything to work except Secure Boot on the UEFI Hyper-V client. And I have downloaded latest wimboot file "I think" version 2.7.2 which should have be signed, right?
> 
> 
> So what´s my issue here? is it because that the ipxe.efi isen't sigend?
> 
> Thanks for this awesome project :D

Hey, you're probably going to need to configure more than just option 67 to get it working for both Legacy and UEFI.  My environment uses iPXE with option 67 and 66 IP Helpers to pass traffic through subnets.  I've also setup DHCP scopes to ID whether it's a Legacy request or UEFI so there's not monumentally large amounts of crosstalk. It's not the 'best practice' method, but it certainly works and it's been holding up for over a year now without any obvious problems.

I basically followed these instructions (I think?) when I set it up.  It's in my bookmarks so I assume it's something I referenced!

https://www.techthoughts.info/pxe-booting-wds-dhcp-scope-vs-ip-helpers/

You're going to want to setup multiple Server Options so you'd have a Legacy and UEFI configuration for Option 66 (should be the same for both as it's just the Server location address AND 67 which would point to the different bootfiles.  

You juggle that by making 2 policies and assign them to the appropriate Server Options.

Policies

UEFI - redirects for UEFI requests 
Legacy - redirects for Legacy requests 

Server Options

066 - Boot Server Host Name - Standard - <servername.domain.com> - UEFI option
067 - Bootfile Name - Standard - boot/ipxe.efi - UEFI option


066 - Boot Server Host Name - Standard - <servername.domain.com> - Legacy option
067 - Bootfile Name - Standard - boot/ipxe.lkrn - Legacy option

Then you breathe into a big paper bag for a while and hope you don't accidentally route all your traffic into a loop and screw everything up.
